eu.lucaweiss.lpa_gtk.yaml

```diff
@@ -0,0 +1,87 @@
+    buildsystem: meson
+    cleanup:
+      - '*'
+    sources:
+      - type: git
+        url: https://gitlab.gnome.org/jwestman/blueprint-compiler
+        tag: v0.16.0
+        x-checker-data:
+          type: git
+          tag-pattern: ^v([\d.]+)$
+        commit: 04ef0944db56ab01307a29aaa7303df6067cb3c0
+  - name: pcsclite # Dependency of lpac
+    config-opts:
+      - --disable-libsystemd
+      - --disable-polkit
+      - --disable-documentation
+    sources:
+      - type: archive
+        url: https://github.com/LudovicRousseau/PCSC/archive/refs/tags/2.3.1.tar.gz
+        sha256: 7fcb59f66a323f63cf1ab492579a57d899806835c52ba377af9ac57df68bf39b
+        x-checker-data:
+          type: anitya
+          project-id: 2611
+          stable-only: true
+          url-template: https://github.com/LudovicRousseau/PCSC/archive/refs/tags/$version.tar.gz
+  - name: lpac # Dependency of lpa-gtk
+    buildsystem: cmake-ninja
+    sources:
+      - type: archive
+        url: https://github.com/estkme-group/lpac/archive/refs/tags/v2.2.1.tar.gz
+        sha256: 3d87080a625b10430eebb82f89e2d24e16a84a8435a9c40b3718fd88c82028ba
+        x-checker-data:
+          type: anitya
+          project-id: 371628
+          stable-only: true
+          url-template: https://github.com/estkme-group/lpac/archive/refs/tags/v$version.tar.gz
+  - name: lpac-logo # As long as lpa-gtk doesn't have it's own
+    sources:
+      - type: file
+        url: https://avatars.githubusercontent.com/u/151148370
+        sha256: e03fa7c49b70397458c0759c78e50db7a755af27c618683b444d589b7000a4a3
+        dest-filename: eSTK_me.jpeg
+    buildsystem: simple
+    build-commands:
+      - for s in {32,64,128,256,512}; do ffmpeg -i eSTK_me.jpeg
+        -s "${s}x${s}" -update true "eSTK_me_${s}x${s}.png";
+        install -Dm644 "eSTK_me_${s}x${s}.png" "${FLATPAK_DEST}/share/icons/hicolor/${s}x${s}/apps/${FLATPAK_ID}.png";
+        done;
+  - name: lpa-gtk
+    sources:
+      - type: git
+        url: https://codeberg.org/lucaweiss/lpa-gtk
+        branch: main
+        commit: a801f87fe1a59c551eb36adbe4189c634226c182
+#     - type: archive
+#       url: https://codeberg.org/lucaweiss/lpa-gtk/archive/0.1.tar.gz
+#       sha256: c55adf95d3029a90a0a62fcce7039d64399cf762870bb44cb2b07122dc61aad1
+#       x-checker-data:
+#         type: json
+#         # Once there are releases:
+#         # url: https://codeberg.org/api/v1/repos/lucaweiss/lpa-gtk/releases/latest
+#         # version-query: .tag_name
+#         # As long as there are no releases:
+#         url: https://codeberg.org/api/v1/repos/lucaweiss/lpa-gtk/tags/
+#         version-query: '.[0].name'
+      - type: patch
+        path: ./desktopfile-icon.patch
+      - type: patch
+        path: ./metainfo-compulsory-release.patch
+      - type: patch
+        path: ./metainfo-screenshot.patch
+    buildsystem: meson
+    build-commands:
+      - mkdir -p ${FLATPAK_DEST}/share/glib-2.0/schemas ${FLATPAK_DEST}/share/icons/hicolor
```

Comment:
why is this necessary?

Reply:
Otherwise `meson` fails:
````
Running custom install script '/usr/bin/glib-compile-schemas /app/share/glib-2.0/schemas'
--- stdout ---

--- stderr ---
Error opening directory “/app/share/glib-2.0/schemas”: No such file or directory

FAILED: install script '/usr/bin/glib-compile-schemas /app/share/glib-2.0/schemas' failed with exit code 1.
FAILED: meson-internal__install 
````

Reply:
buildsystem and build-commands won't work together. If you use a named buildsystem, build-commands come from that.

Reply:
OK, I have raised a PR upstream about the schema compilation.
The desktop-file-edit should not be needed once either https://gitlab.gnome.org/Teams/Design/app-icon-requests/-/issues/88 or an equivalent is done.

Reply:
> As for the desktop-file-edit it should be in post-install

Apparently `appstreamcli compose` runs before `post-install` and it fails with `gui-app-without-icon`, so trying to edit the icon property in `post-install` appears to be too late.

Reply:
> Apparently appstreamcli compose runs before post-install

no it doesn't.

Reply:
> no it doesn't.

Right. Now I see.

> trying to edit the icon property in post-install appears to be too late.

What was indeed too late is to still edit the desktop file under `/run/build/`. In the `post-install` the desktop file is better edited under `/app/share/applications/`.